OpenEdXMobile/src/main/java/org/edx/mobile/view/MainDiscoveryFragment.java

```diff
@@ -221,16 +224,27 @@ public void onFragmentVisibilityChanged(final boolean isVisibleToUser) {
                 @Override
                 public void run() {
                     final Fragment nativeCoursesFragment = getChildFragmentManager().findFragmentByTag("fragment_courses_native");
-                    if (nativeCoursesFragment != null && nativeCoursesFragment.isVisible()) {
+                    if ((nativeCoursesFragment != null && nativeCoursesFragment.isVisible()) || !isVisibleToUser) {
                         toolbarCallbacks.getSearchView().setVisibility(View.GONE);
                     } else {
-                        toolbarCallbacks.getSearchView().setVisibility(isVisibleToUser ? View.VISIBLE : View.GONE);
+                        updateDiscoverFragmentsVisibility();
                     }
                 }
             });
         }
     }
 
+    private void updateDiscoverFragmentsVisibility() {
```

Comment:
`updateShowFragmentsVisbility` would be a better name for this function. Thoughts @farhan?

Also, this function could use a brief Javadoc.

Reply:
> `updateShowFragmentsVisbility` would be a better name for this function. Thoughts @farhan?
> 
`updateShowFragmentsVisbility` name seems ambiguous. Like we are updating show fragment visibility.

Reply:
Actually I made a typo, the name should've been `updateShownFragmentsVisbility`.

Which'll be read as `Update shown fragment's visibility`.